tools/java/cpp-build/src/com/google/i18n/phonenumbers/BuildMetadataCppFromXml.java

```diff
@@ -201,7 +203,16 @@ public boolean start() {
   // @VisibleForTesting
   void writePhoneMetadataCollection(
       String inputFilePath, boolean liteMetadata, OutputStream out) throws IOException, Exception {
-    BuildMetadataFromXml.buildPhoneMetadataCollection(inputFilePath, liteMetadata).writeTo(out);
+    // The size of the byte buffer used for serializing the PhoneMetadataCollection.
+    int COLLECTION_BUFFER_SIZE = 256 * 1024;
+
+    PhoneMetadataCollection metadataCollection =
+        BuildMetadataFromXml.buildPhoneMetadataCollection(inputFilePath, liteMetadata);
+    byte[] outputArray = new byte[COLLECTION_BUFFER_SIZE];
+    CodedOutputByteBufferNano outputByteBuffer = CodedOutputByteBufferNano.newInstance(outputArray);
+    metadataCollection.writeTo(outputByteBuffer);
```

Comment:
Should have a helper function that takes care of the CodedOutputByteBufferNano; e.g.:
```
writePhoneMetadataCollection(PhoneMetadataCollection, OutputStream);
```